### PR TITLE
Feature/add dagre autolayout hook for demo

### PR DESCRIPTION
**Pull request by ChTiSh**
## Description
Add dagre auto layout hook for demo
!!!This will not work with nodes with parentId!!

Fixes # (issue)

## Type of change

- [ ] 🐛 Bug fix (non-breaking change which fixes an issue)
- [ ] ✨ New feature (non-breaking change which adds functionality)
- [ ] 💥 Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] 📝 Documentation update
- [ ] Other 🗒️ ______________

## How Has This Been Tested?

1. import useAutoLayout hook to the Flow.tsx
2. setup dummy data with positions all being {0, 0}
3.  call the hook with sample nodes, edges, width and height

## Checklist:

- [ ] I have performed a self-review of my own code.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] My changes generate no new warnings.
- [ ] Any dependent changes have been merged and published in downstream modules.
- [ ] I have reviewed the [**Security Best Practices Document**](./docs/SECURITY_BEST_PRACTICES.md).
- [ ] I have followed the [**Naming Conventions Guide**](./docs/NAMING_CONVENTIONS_GUIDE.md) for my changes.
- [ ] I have ensured that my pull request title is descriptive.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -62,6 +63,7 @@
     "axios": "^1.6.7",
     "cookie-parser": "^1.4.6",
     "cors": "^2.8.5",
+    "dagre": "^0.8.5",
```
Comment (relandboyle) [inline package.json]: @ChTiSh Will you clarify - Do we need both dagre and @dagrejs/dagre libraries?

File: package-lock.json
Comment (relandboyle) [inline package-lock.json:1]: @ChTiSh It looks like you deleted your package-lock.json and then pushed changes before running npm install. Maybe it makes sense to generate a new file before merging this PR?

image

File: package.json (hunk 1 above)
Comment (kinskiwu) [inline package.json]: This looks like the old npm package.

**Comment by ChTiSh** [comment]
PR closed due to recent discussion of layout algorithms.

**Comment by relandboyle** [comment]
@ChTiSh I'm just noticing this PR was closed. I would like to see it reopened, updated, and merged with the Dev branch.

This PR probably should have been resolved on Friday 05/10 when its contents were included in a demo for the team. Not merging, and ultimately closing, may lead to confusion and ambiguity about what work was completed and what product version was demonstrated.

**Comment by ChTiSh** [comment]
Hi @relandboyle I thought the discussion led to we should implement our algo for layout, thus this implement using dagre library is not valid anymore, so I closed it to avoid confusion. I can reopen it if it could be useful

**Comment by relandboyle** [comment]
@ChTiSh I think this discussion is important as it may apply to any number of future decisions.

It's fine if a decision was made to develop custom algorithms rather than using the Dagre library. I'm sure we can do that. But we have not done it yet. It may be weeks or months before we start that dev effort. Even if it's only a few days in the future, as long as it is in the future there is no reason to break the present solution.

Dagre was included in a demo on 05/10. Arguably, it should have been reviewed and merged before the demo. The demo audience should expect to find Dagre in the repo as of the time of the demo. Removing or omitting Dagre means that developers may run into an unexpected, less-functional version of the application. As of now, there have been six days when the Dev branch is effectively behind the latest known Sandbox version of the application, and as far as I know the custom algorithms have not been implemented. I feel that we should never allow this condition to occur.

**Review by relandboyle — APPROVED** [review]

**Pull request merged**